mmv1/products/documentaiwarehouse/DocumentSchema.yaml

```diff
@@ -0,0 +1,362 @@
+          description: |
+            The display-name for the property, used for front-end.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isRepeatable'
+          description: |
+            Whether the property can have multiple values.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isFilterable'
+          description: |
+            Whether the property can be filtered. If this is a sub-property, all the parent properties must be marked filterable.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isSearchable'
+          description: |
+            Indicates that the property should be included in a global search.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isMetadata'
+          description: |
+            Whether the property is user supplied metadata.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isRequired'
+          description: |
+            Whether the property is mandatory.
+        - !ruby/object:Api::Type::Enum
+          name: 'retrievalImportance'
+          description: |
+            Stores the retrieval importance.
+          values:
+            - :HIGHEST
+            - :HIGHER
+            - :HIGH
+            - :MEDIUM
+            - :LOW
+            - :LOWEST
+        - !ruby/object:Api::Type::Array
+          name: 'schemaSources'
+          description: |
+            The schema source information.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'name'
+                description: |
+                  The schema name in the source.
+              - !ruby/object:Api::Type::String
+                name: 'processorType'
+                description: |
+                  The Doc AI processor type name.
+        # select any one of the property types (integer, float, text, property, enum, dateTime, map, timestamp)
+        - !ruby/object:Api::Type::NestedObject
+          name: 'integerTypeOptions'
+          description: |
+            Integer property.
+          send_empty_value: true
+          allow_empty_object: true
+          properties: []
+        - !ruby/object:Api::Type::NestedObject
+          name: 'floatTypeOptions'
+          description: |
+            Float property.
+          send_empty_value: true
+          allow_empty_object: true
+          properties: []
+        - !ruby/object:Api::Type::NestedObject
+          name: 'textTypeOptions'
+          description: |
+            Text/string property.
+          send_empty_value: true
+          allow_empty_object: true
+          properties: []
+        - !ruby/object:Api::Type::NestedObject
+          name: 'propertyTypeOptions'
+          description: |
+            Nested structured data property.
+          send_empty_value: true
+          properties:
+            - !ruby/object:Api::Type::Array
+              name: 'propertyDefinitions'
+              description: |
```

Comment:
It looks like `required: true` is needed for the field 'propertyDefinitions', as it is required based on the API doc